pkg/downloader/manager.go

```diff
@@ -530,6 +530,11 @@ func (m *Manager) ensureMissingRepos(repoNames map[string]string, deps []*chart.
 	// the dependencies that are not known to the user if update skipping
 	// is not configured.
 	if !m.SkipUpdate && len(ru) > 0 {
+
+		// Some dependencies might have a different names but point to the same repository
```

Comment:
It's necessary because the current way it works now, it only validate the dependencies name:
```golang
		// When the repoName for a dependency is known we can skip ensuring
		if _, ok := repoNames[dd.Name]; ok {
			continue
		}
```

To illustrate, assume you have the following dependencies:
```golang
[]*chart.Dependency{
		&chart.Dependency{
			Name: "dep",
			Repository: "somerepo",
		},
		&chart.Dependency{
			Name: "anotherDep",
			Repository: "somerepo",
		},
	}
```
this would result in the following entries:
```golang
[]*repo.Entry{
		&repo.Entry{
			Name: "helm-manager-b943c55ec6eb8d857609e556ebfa9683f499f1f97d6758195bab3dc210373fd6",
			URL: "somerepo",
		},
		&repo.Entry{
			Name: "helm-manager-b943c55ec6eb8d857609e556ebfa9683f499f1f97d6758195bab3dc210373fd6",
			URL: "somerepo",
		},
	}
```
As you can see, there's duplication and would be passed to `parallelRepoUpdate` function